### PR TITLE
[#348] Move ProxyVerificationKey to Delegation.Certificate

### LINKED ISSUE #348: Simplify proxy signing
After #347 the `ProxySecretKey` will be called `ProxyVerificationKey`. This is being used as a delegation certificate and nothing else, but is designed as a more general scheme. I think we should remove this premature abstraction, so I propose to:
- Remove the `omega` parameter and concretise to `EpochIndex`
- Inline the definitions from `ProxySecretKey` to `Delegation.Certificate`
- Perhaps change the name to `Delegation.OperationalCertificate` for forwards compatibility with Shelley

### PR DESCRIPTION
**Pull request by ruhatch**
This moves the `ProxyVerificationKey` from `cardano-crypto-wrapper` into `cardano-ledger`, as it is only ever used as a `Delegation.Certificate`.

Closes #348

### CODE REVIEW

**Review by intricate — APPROVED** [review]
This definitely makes more sense being referred to as a `Certificate`. ~However, shouldn't we also remove `AProxyVerificationKey` from `cardano-crypto-wrapper`?~

EDIT: Oops, didn't notice that you actually did remove it :+1:

File: cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs [hunk 1]
```diff
@@ -1,45 +1,204 @@
...
+--------------------------------------------------------------------------------
+-- Certificate
+--------------------------------------------------------------------------------
 
--- | A delegation certificate is a `ProxyVerificationKey` tagged with an `EpochIndex`
 type Certificate = ACertificate ()
 
-type ACertificate a = AProxyVerificationKey EpochIndex a
+-- | Delegation certificate allowing the @delegateVK@ to sign blocks on behalf
+--   of @issuerVK@, from epoch @aEpoch@ until a new delegation is activated
```
Comment (dnadales) [inline cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs]: BTW, in the Byron rules we don't check that the activation epoch corresponds with the epoch in the certificate. Epochs are used for replay protection and protection against resource exhaustion only. Is this ok?

**Review by dnadales — APPROVED** [review]

File: cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs (hunk 1 above)
Comment (ruhatch) [reply cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs]: Yeah, good point, I'll adjust this comment

**Pull request merged**